yivi-docs/blog/2025-09-25-openid4vp/index.md

```diff
@@ -0,0 +1,180 @@
+the classic Idemix format, as well as the new SD-JWT VC format using the latest version of the `irma` server.
+
+In order to be able to do this they need to obtain a new x509 certificate that's on our new Trust List.
+
+### Keybinding & salts: A major limitation
+In order to prove a credential instance belongs to a user, SD-JWT uses a system called "key binding".
+Here the wallet provides an asymetric public key to the issuer, who puts it into the credential.
+During disclosure, the wallet uses its corresponding private key to sign a JWT containing the hash over the credential and some
+parameters provided by the verifier to prove they own the private key corresponding to the public key inside of the credential.
+
+This is quite a nice system, but it has one major flaw:
+disclosing a credential multiple times means showing the same public key multiple times.
+This breaks one of the pilars Yivi was originally built on: multi show unlinkability.
+
+The only way to (kind of) solve this without moving entirely to Zero knowledge Proofs (like Yivi originally uses),
+is to issue multiple instances of the same credential at the same time and throwing an instance away after it has been used once.
+
+This has far reaching consequences for the user experience,
+which is something we'll talk about a next.
+
+
+## The Yivi app
+For our users not a lot will change in terms of how the Yivi app works.
+This is intentional, as we wanted to stay close to our easy to understand interface.
+The issuance and disclosure flows, including the ability to do issuance during discosure, remain unchanged.
+
+One thing we did change is the way credential cards look.
+This has to do with the fact that SD-JWT VCs come in batches of instances (as discussed above) 
+and these instances are removed after a single use.
+If you received 10 instances you can share it only 10 times.
+We struggled with this, because the implications for the UX are quite big.
+
+As you might know, Yivi's own Idemix credentials are infinitely sharable, until they are expired based on time.
+Issuing the same credential in SD-JWT means that the app possess one format that can be shared infinitely,
+and another format that can only be shared X times.
+
+This leaves us in a situation where a credential can be in a state of both valid and "out of stock",
+depending on which format is requested.
+
+We don't want to bother our users with this, but in one way or another we'll have to.
+
+After a lot of consideration we decided to show the instance count in the UI.
+It now shows up next to the expiry date. If it's Idemix only we show "unlimited", if an SD-JWT is present, we show the remaining number of
+disclosures.
+
+When the credential is close to its usage limit, we show this to the user with a button to reobtain the credential,
+just like we do with regular time-based expiry.
+
+This leaves us with only two UX downsides:
```

Comment:
The app could show that SD-JWT VC are depleted but Idemix credentials are still available.
The app doesn't show a decreased number when Idemix credentials are disclosed, which might be strange to the user.